pandas/io/parsers.py

```diff
@@ -801,6 +803,15 @@ def _clean_options(self, options, engine):
                                   " different from '\s+' are"\
                                   " interpreted as regex)"
                 engine = 'python'
+            elif quoting != csv.QUOTE_NONE:
```

Comment:
my point is that ``\s+`` is NOT a multi-char separator to the c-engine, it is instead passed as the ``delim_whitespace`` kw. So quoting is properly handled. Not sure this should be different in the python parser.